src/lib.rs

```diff
@@ -22,12 +31,49 @@
     trivial_numeric_casts,
     unreachable_pub,
     unused,
-    variant_size_differences
+    variant_size_differences,
+
+    clippy::pedantic,
+    clippy::nursery,
+    clippy::suspicious,
+    clippy::complexity,
+    //clippy::cargo,
+
+    // Lints from clippy::restriction
+    clippy::string_add, // Don't use + operator to merge strings
+    clippy::string_to_string, // String::to_string() -> String::clone()
+    clippy::unneeded_field_pattern, // Match only what's needed on struct
+    clippy::rest_pat_in_fully_bound_structs, // Don't use .. in a fully bound struct
+    clippy::unnecessary_self_imports, // use std::io::{self} -> use std::io
+    clippy::get_unwrap, // arr[0] instead of arr.get(0).unwrap()
+    clippy::clone_on_ref_ptr, // Don't clone a pointer
+    clippy::create_dir, // std::fs::create_dir() -> std::fs::create_dir_all()
+    clippy::filetype_is_file, // std::fs::is_file -> !std::fs::is_dir()
+    clippy::if_then_some_else_none, // Use bool::then() instead of an if block that returns Option
+    //clippy::default_numeric_fallback, // Warn when compiler can't infer type and will use a fallback type
+    //clippy::integer_division, // Use floats when dividing or you might lose precision!
+    clippy::lossy_float_literal, // Warn when float literal might lose precision
+    clippy::str_to_string, // str.to_string() -> str.to_owned()
+    clippy::unseparated_literal_suffix, // Separate the integer and its type with underscore: 6_usize
+)]
+#![allow(
+    // 🦀 Modules repeat A LOT, but I don't want to change so much if it's not a welcome change. Do
+    // you think it's reasonable to change modules' names?
+    clippy::module_name_repetitions,
+
+    // 🦀 These lints I've disabled since there are so much of them! And the worst part is that
+    // most of the time I can't tell if I'll break something if I try to make changes according to
+    // those lints, so I kind of need help here :/
+    clippy::as_conversions,
```

Comment:
Again - these lints are trying to guard against unexpected usage, but if you read the lint docs - they indicate there are a lot of valid use cases for these conversions and as I mentioned in my main comment - I did a large pass through of many of these lints a few weeks back. I'm aware of all my use cases of losing a signed integer or truncating (and in many cases you can see I specifically round() floats before casting them to indicate my intent).

Sadly there is no way to do many types of casts and still satisfy these lints.